var/spack/repos/builtin/packages/python/package.py

```diff
@@ -454,7 +454,12 @@ def get_config_var(self, key):
         cmd = 'from distutils.sysconfig import get_config_var; '
         cmd += self.print_string("get_config_var('{0}')".format(key))
 
-        return self.command('-c', cmd, output=str).strip()
+        # Spack-installed Python works best
+        # without other env vars polluting it.
+        # https://askubuntu.com/questions/640010
+        pythoncmd = self.command('-c', cmd, output=str).strip()
+        pythoncmd.add_default_env('PYTHONPATH', '')
+        return pythoncmd
```

Comment:
`pythoncmd` is a string, you need to add the environment to the `Executable`. Can you try:
```python
self.command.add_default_env('PYTHONPATH', '')
return self.command('-c', cmd, output=str).strip()
```